File: tests/e2e/cases/09-type-coverage/bool_or_to_u32/bool_or_to_u32_disproved.rs
// DEMO: BUGGY bool-or-to-u32 — uses AND instead of OR.
// SAW disproves with a=true, b=false.

fn bool_or_to_u32(a: bool, b: bool) -> u32 {
    if a && b { 1 } else { 0 }   // BUG: was a || b.
}

File: tests/e2e/cases/09-type-coverage/bool_or_to_u32/bool_or_to_u32_disproved.cpp
/*
DEMO: BUGGY bool-or-to-u32 — uses AND instead of OR.

SAW disproves with a=true, b=false (impl returns 0, spec says 1).

Expected verdict: DISPROVED.
*/

#include <cstdint>

uint32_t bool_or_to_u32(bool a, bool b) {
    return (a && b) ? 1u : 0u;   // BUG: was a || b.
}
